dali/imgcodec/decoders/nvjpeg/nvjpeg.cc

```diff
@@ -171,17 +173,52 @@ DecodeResult NvJpegDecoderInstance::DecodeImplTask(int thread_idx,
   } else {
     CUDA_CALL(nvjpegDecodeParamsSetROI(ctx.resources.params, 0, 0, -1, -1));
   }
+
+  auto orientation = opts.use_orientation ? JpegParser().GetInfo(in).orientation : Orientation{};
+  bool is_orientation_adjusted = orientation.rotate || orientation.flip_x || orientation.flip_y;
+
+  // We don't decode directly to YCbCr, since we want to control the YCbCr definition,
+  // which is different between general color conversion libraries (OpenCV) and
+  // what JPEG uses.
+  // JPEG files are always using bitdepth 8.
+  bool needs_processing = opts.format == DALI_YCbCr ||
+                          opts.dtype != DALI_UINT8 ||
+                          is_orientation_adjusted;
+
   try {
     ParseJpegSample(*in, opts, ctx);
+
     if (roi.use_roi()) {
       ctx.shape[0] = roi.shape()[0];
       ctx.shape[1] = roi.shape()[1];
     }
-    DecodeJpegSample(*in, out.mutable_data<uint8_t>(), opts, ctx);
+
+    // Synchronizing on the acces to immediate_buffer
+    auto& intermediate_buffer = ctx.resources.intermediate_buffer;
+    CUDA_CALL(cudaEventSynchronize(ctx.resources.decode_event));
```

Comment:
Only when the buffer is actually used. So I guess here It could be conditional, and later on when some other resources are used we can sync again? The second sync with be a no-op, right?

Reply:
The synchronization would have to happen almost in the same place anyways (https://github.com/NVIDIA/DALI/pull/4247), I don't think it makes any difference.